src/Microsoft.Management.Deployment/PackageCatalogReference.cpp

```diff
@@ -22,33 +22,19 @@ namespace winrt::Microsoft::Management::Deployment::implementation
 {
     void PackageCatalogReference::Initialize(winrt::Microsoft::Management::Deployment::PackageCatalogInfo packageCatalogInfo, ::AppInstaller::Repository::Source sourceReference)
     {
-        static constexpr winrt::Windows::Foundation::TimeSpan s_PackageCatalogUpdateIntervalDelay = 168h; //1 week
-
         m_info = packageCatalogInfo;
         m_sourceReference = std::move(sourceReference);
         m_packageCatalogBackgroundUpdateInterval = ::AppInstaller::Settings::User().Get<::AppInstaller::Settings::Setting::AutoUpdateTimeInMinutes>();
 
-        bool shouldDelayBackgroundUpdateInterval = false;
-        try
+        if (ShouldDelayBackgroundUpdateByCaller())
         {
-            auto [hrGetCallerId, callerProcessId] = GetCallerProcessId();
-            THROW_IF_FAILED(hrGetCallerId);
-            if (callerProcessId != GetCurrentProcessId())
-            {
-                // OutOfProc case, we check for explorer.exe
-                auto callerNameWide = AppInstaller::Utility::ConvertToUTF16(GetCallerName());
-                auto processName = AppInstaller::Utility::ConvertToUTF8(std::filesystem::path{ callerNameWide }.filename().wstring());
-                if (::AppInstaller::Utility::CaseInsensitiveEquals("explorer.exe", processName))
-                {
-                    shouldDelayBackgroundUpdateInterval = true;
-                }
-            }
-        }
-        CATCH_LOG();
+            static constexpr winrt::Windows::Foundation::TimeSpan s_PackageCatalogUpdateIntervalDelay_Base = 168h; //1 week
 
-        if (shouldDelayBackgroundUpdateInterval)
-        {
-            m_packageCatalogBackgroundUpdateInterval = s_PackageCatalogUpdateIntervalDelay;
+            // Add a bit of randomness to the default interval time
+            std::default_random_engine randomEngine(std::random_device{}());
+            std::uniform_int_distribution<long long> distribution(0, 168);

```

Comment:
We really want to spread this out, so multiple this by 3600 for second resolution.